applications/rag/frontend/main.tf

```diff
@@ -30,6 +34,43 @@ locals {
     instance_connection_name = format("%s:%s:%s", var.project_id, var.region, "pgvector-instance")
 }
 
+# IAP Section: Enabled the IAP service
+resource "google_project_service" "project_service" {
+  project = var.project_id
+  service = "iap.googleapis.com"
+
+  disable_dependent_services = false
+  disable_on_destroy         = false
+}
+
+# IAP Section: Creates the OAuth client used in IAP
+resource "google_iap_client" "iap_oauth_client" {
+  count        = var.client_id == "" ? 1 : 0
+  display_name = "Frontend-Client"
```

Comment:
does this need to be a separate client? can jupyterhub and the frontend share a client for RAG, to make the setup simpler for the user?

I guess there could be scenarios where a developer would only want to grant access to the frontend chat bot, though.